hubbard/plot/spectrum.py

```diff
@@ -149,36 +144,114 @@ def __init__(self, HubbardHamiltonian, k=[0, 0, 0], spin=0,
-        self.set_title(title)
+
         self.set_xlabel(r'distance along [%.2f,%.2f,%.2f] (\AA)' % tuple(direction))
         self.set_ylabel(r'$E-E_\mathrm{midgap}$ (eV)')
         self.set_xlim(xmin, xmax)
         self.set_ylim(emin, emax)
         self.axes.set_aspect('auto')
 
 
-class DOS_distribution(GeometryPlot):
+class EigenLDOS(GeometryPlot):
     """ Plot LDOS in the configuration space for the `hubbard.HubbardHamiltonian` object
 
     Parameters
     ----------
     HH: HubbardHamiltonian
         mean-field Hubbard Hamiltonian
-    PDOS: numpy.ndarray
-        Density of states projected in atomic sites obtained at certain energy
+    WF: numpy.ndarray or sisl.physics.electron.EigenstateElectron
+        Eigenstate to be plotted as LDOS, WF can be an array-like containing more than WF
+        The WF have to be stored along the second axis, i.e. each WF is stored as column vectors
+        in this case this method sums the EigenLDOS corresponding to each WF in the grid
+        This can be handy if one wants to plot the LDOS = LDOS_up + LDOS_dn
     realspace:
-        If True it will plot the DOS in a realspace grid otherwise it plots it as a scatter plot
+        If True it will plot the LDOS in a realspace grid otherwise it plots it as a scatter plot (PDOS)
+        with varying size depending on the PDOS numerical value
+    """
+
+    def __init__(self, HH, WF, sites=[], ext_geom=None, realspace=False, **kwargs):
+
+        # Set default kwargs
+        if realspace:
+            if 'facecolor' not in kwargs:
+                kwargs['facecolor'] = 'None'
+            if 'cmap' not in kwargs:
+                kwargs['cmap'] = 'Greys'
+        else:
+            if 'cmap' not in kwargs:
+                kwargs['cmap'] = plt.cm.bwr
+
+        super().__init__(HH.geometry, ext_geom=ext_geom, **kwargs)
+
+        x = HH.geometry[:, 0]
+        y = HH.geometry[:, 1]
+
+        # Ensure WF is an array of correct dimensions
+        WF = np.array(WF)
+        if WF.shape == (len(WF),):
+            WF = np.expand_dims(WF, axis=1)
+
+        if realspace:
+            if 'grid_unit' not in kwargs:
+                kwargs['grid_unit'] = [100,100,1]
+            if 'z' not in kwargs:
+                kwargs['z'] = 1.1
+
+            if 'vmin' not in kwargs:
+                kwargs['vmin'] = 0
+
+            xmin, xmax, ymin, ymax = self.xmin, self.xmax, self.ymin, self.ymax
+
+            grid = 0
+            for i in range(WF.shape[1]):
+                wf = WF[:,i]
+                grid_i = real_space_grid(self.geometry, wf, kwargs['grid_unit'], xmin, xmax, ymin, ymax, z=kwargs['z'], mode='wavefunction')
+                grid += grid_i ** 2
+
+            self.__realspace__(grid, **kwargs)
+            self.imshow.set_cmap(plt.cm.afmhot)
+
+        else:
+            wf = 0
+            for i in range(WF.shape[1]):
+                v = WF[:,i]
+                wf += v ** 2
```

Comment:
Not sure it is a good idea to overwrite wf with probability density. It is no longer the same thing.

Reply:
It wasn't overwriting it since `wf` is a different variable from `WF`. But I agree that is confusing, I will change the name of `wf` to e.g. `dos`, which is more representative.